src/core/thread/mle.cpp

```diff
@@ -1713,43 +1713,70 @@ void Mle::HandleAttachTimer(Timer &aTimer)
     aTimer.Get<Mle>().HandleAttachTimer();
 }
 
-void Mle::HandleAttachTimer(void)
+bool Mle::HasAcceptableParentCandidate(void) const
 {
-    uint32_t delay          = 0;
-    bool     shouldAnnounce = true;
+    bool hasAcceptableParent = false;
 
-    if (mAttachState == kAttachStateParentRequestRouter || mAttachState == kAttachStateParentRequestReed ||
-        (mAttachState == kAttachStateAnnounce && !HasMoreChannelsToAnnouce()))
+    switch (mAttachState)
     {
-        uint8_t linkQuality;
+    case kAttachStateAnnounce:
+        VerifyOrExit(!HasMoreChannelsToAnnouce());
+        OT_FALL_THROUGH;
 
-        linkQuality = mParentCandidate.GetLinkInfo().GetLinkQuality();
+    case kAttachStateParentRequestRouter:
+    case kAttachStateParentRequestReed:
+        break;
 
-        if (linkQuality > mParentCandidate.GetLinkQualityOut())
-        {
-            linkQuality = mParentCandidate.GetLinkQualityOut();
-        }
+    default:
+        ExitNow();
+    }
+
+    VerifyOrExit(mParentCandidate.IsStateParentResponse());
```

Comment:
nit: move above switch statement?

Reply:
Thanks. Great suggestion. 👍 
With this we can move the link quality check to the `switch` as well (makes it simpler).  

Changed them both in the new push.